keras/src/export/litert.py

```diff
@@ -0,0 +1,255 @@
+        ):
+            return {
+                name: make_input_spec(inp)
+                for name, inp in self.model._inputs_struct.items()
+            }
+
+        # Fall back to model.inputs if it's a dict
+        if hasattr(self.model, "inputs") and isinstance(
+            self.model.inputs, dict
+        ):
+            return {
+                name: make_input_spec(inp)
+                for name, inp in self.model.inputs.items()
+            }
+
+        return None
+
+    def export(self, filepath):
+        """Exports the Keras model to a TFLite file.
+
+        Args:
+            filepath: Output path for the exported model
+
+        Returns:
+            Path to exported model
+        """
+        # 1. Resolve / infer input signature
+        if self.input_signature is None:
+            # Try dict-specific inference first (for models with dict inputs)
+            dict_signature = self._infer_dict_input_signature()
+            if dict_signature is not None:
+                self.input_signature = dict_signature
+            else:
+                # Fall back to standard inference
+                self.input_signature = get_input_signature(self.model)
+
+        # 3. Handle dictionary inputs by creating an adapter
+        # Check if we have dict inputs that need adaptation
+        has_dict_inputs = isinstance(self.input_signature, dict)
+
+        if has_dict_inputs:
+            # Create adapter model that converts list to dict
+            adapted_model = self._create_dict_adapter(self.input_signature)
+
+            # Convert dict signature to list for TFLite conversion
+            # The adapter will handle the dict->list conversion
+            input_signature_list = list(self.input_signature.values())
+
+            # Use adapted model and list signature for conversion
+            model_to_convert = adapted_model
+            signature_for_conversion = input_signature_list
+        else:
+            # No dict inputs - use model as-is
+            model_to_convert = self.model
+            signature_for_conversion = self.input_signature
+
+        # Store original model reference for later use
+        original_model = self.model
+
+        # Temporarily replace self.model with the model to convert
+        self.model = model_to_convert
+
+        try:
+            # 4. Convert the model to TFLite.
+            tflite_model = self._convert_to_tflite(signature_for_conversion)
+        finally:
+            # Restore original model
+            self.model = original_model
+
+        # 4. Save the initial TFLite model to the specified file path.
+        if not filepath.endswith(".tflite"):
+            raise ValueError(
+                "The LiteRT export requires the filepath to end with "
+                "'.tflite'. Got: {filepath}"
+            )
+
+        with open(filepath, "wb") as f:
+            f.write(tflite_model)
+
+        return filepath
+
+    def _create_dict_adapter(self, input_signature_dict):
```

Comment:
Let's rename it since it's for more than just dicts. `_create_nested_inputs_adapter(self, input_signature_struct)`.